ekncontent/ekncontent/eknc-domain.c

```diff
@@ -299,9 +307,14 @@ eknc_domain_initable_init (GInitable *initable,
   g_autoptr(GFile) bundle_subscriptions_dir = g_file_get_child (self->content_dir, "com.endlessm.subscriptions");
   g_autoptr(GFile) bundle_dir = g_file_get_child (bundle_subscriptions_dir, self->subscription_id);
 
-  if (!g_file_query_exists (subscription_dir, cancellable) &&
-      !g_file_make_directory_with_parents (subscription_dir, cancellable, error))
-      return FALSE;
+  g_file_make_directory_with_parents (subscription_dir, cancellable, error);
+  if (*error)
+    {
+      if (g_error_matches (*error, G_IO_ERROR, G_IO_ERROR_EXISTS))
```

Comment:
This will crash if the caller of this function passes `NULL` in to indicate they aren't interested in the error. I think that's the reason for the `local_error` pattern.